### PR TITLE
add instruction for self-hosted SAML Authentication Options

### PR DESCRIPTION
**Pull request by iamtodor**
Hello,
This PR brings huge documentation improvement and clarification for SSO SAML Authentication.
PR has been established based on the opened issue https://github.com/getredash/redash/issues/2977
Kudos to @rockxsj

### CODE REVIEW

File: src/pages/kb/user-guide/users/authentication-options.md [hunk 1]
```diff
@@ -10,61 +10,96 @@ Authentication settings are available in the `Settings`>`Settings` tab.
...
+possible to configure the SAML provider to pass what groups the user should join
+by setting the `RedashGroups` parameter. If you use OneLogin's predefined Redash
+application, it will always pass this parameter, meaning that even for existing
+users, it will override their current groups memberships. Hence you need to make
+sure it's up to date.
+
+### How to Configure self-hosted SAML
+
+1. SAML Metadata URL has to look as following:
+   `http://your-site.com/auth/realms/somerelm/protocol/saml/descriptor`
```
Comment (susodapop) [inline src/pages/kb/user-guide/users/authentication-options.md:51]: This isn't strictly true. The metadata URL should point at an XML file. Perhaps update to the following?

```suggestion
1. SAML Metadata URL should point to an XML file on your server, such as:
   `http://your-site.com/auth/realms/somerelm/protocol/saml/descriptor`
```

File: src/pages/kb/user-guide/users/authentication-options.md [hunk 2]
```diff
@@ -10,61 +10,96 @@ Authentication settings are available in the `Settings`>`Settings` tab.
...
+2. SAML Entity ID has to be: `redash`
```
Comment (susodapop) [inline src/pages/kb/user-guide/users/authentication-options.md:52]:
```suggestion
2. SAML Entity ID should be: `redash`
```

File: src/pages/kb/user-guide/users/authentication-options.md [hunk 3]
```diff
@@ -10,61 +10,96 @@ Authentication settings are available in the `Settings`>`Settings` tab.
...
+3. SAML NameID Format has to be:
```
Comment (susodapop) [inline src/pages/kb/user-guide/users/authentication-options.md:53]:
```suggestion
3. SAML NameID Format should be:
```

**Review by susodapop — CHANGES_REQUESTED** [review]
A couple small changes and we can merge this.

Thanks!

**Comment by iamtodor** [comment]
@susodapop Thank you for the changes! I committed them

**Review by susodapop — APPROVED** [review]
LGTM

**Comment by iamtodor** [comment]
@susodapop awesome, thank you! :)

**Pull request merged**